specification/data-resource-semantic-conventions.md

```diff
@@ -70,6 +70,28 @@ Labels defining a compute unit (e.g. Container, Process, Lambda Function).
 | container.image.name | Name of the image the container was built on. | `gcr.io/opentelemetry/operator` |
 | container.image.tag | Container image tag. | `0.1` |
 
+
+### Function as a Service
+
+**type:** `faas`
+
+**Description:** A serverless instance.
+
+| Label  | Description  | Example  | Required |
+|---|---|---|--|
+| faas.region| A specific geographical location where the function is executed. | `us-east-2` | Yes |
+| faas.name | The name of the function being executed. | `opentelemetry-lambda` | Yes |
+| faas.identifier | The unique name of the function being executed. <br /> For example, in AWS this field correspond to the [ARN] value, in GCP to the URI of the resource, and Azure to the [FunctionDirectory] field. | `opentelemetry-lambda` | Yes |
+| faas.version | The version of the function being executed. | `opentelemetry-latest` | No |
+| faas.provider | Provider name of the function, e.g. `aws`, `gpc`, `azure`, `serverless`, ... | `aws` | No |
+| faas.payload | The input passed to the function. | `{name: 'opentelemetry'}` | No |
```

Comment:
This description leaves a lot of room for interpretation. For instance, on AWS Lambda triggered by HTTP, is this the whole `event` object or just the POST body?

Reply:
This could get very big potentially.
Also, we have to consider masking data here (i.e. make sure the server does it)

Reply:
Agree, I should make it more specific.

Reply:
> This could get very big potentially.
> Also, we have to consider masking data here (i.e. make sure the server does it)

On a second thought, is this field really important? Having this field might be dangerous, from privacy and resource required POV

Reply:
True. There was talk about having "enhanced information"/"discarded by default" fields here: https://github.com/open-telemetry/opentelemetry-specification/pull/397#discussion_r367966835 But for now I think it would be best to leave the field out.